tests/test_fed_avg_optimizer.py

```diff
@@ -109,3 +118,10 @@ def test_optimizer_schedules_communication_after_training(initialization_payload
     assert event_type == ActionableEventTypes.SCHEDULE_JOBS.name
     for job in job_arr:
         assert job.job_type == JobTypes.JOB_COMM.name
+
+def test_cleanup():
```

Comment:
Although convenient since it runs after all of the tests, I wouldn't make clean up a test since it's just something that needs to be done rather than a functionality of the Optimizer you're actually testing.

Tl;dr s/test_cleanup/cleanup, remove the assert statement, and call this method at the end of the relevant tests. If this seems repetitive, you might want to consider using something similar to teardown functionality, see [here](https://stackoverflow.com/questions/22627659/run-code-before-and-after-each-test-in-py-test)

Reply:
I had the exact functionality you're describing before. It didn't work since tests that failed wouldn't get to their cleanup code, which means that developers have to rm the transformed data every time. I will look into using the fixture.

Reply:
Fixture doesn't work. I'm going to leave as is until an actual solution can be found.

Reply:
That's a good point (I've had the same issue), but I still think that it shouldn't be a test when you're merging. Perhaps you can do clean up at end of methods and make this function available in testing utils. That way, when people start developing they can just copy over the general function, modify for their usage, and then delete for PR.

If you still disagree, then let's make this a card and figure it out later.

Reply:
Did your thing